src/topology/test/mod.rs

```diff
@@ -140,13 +141,15 @@ async fn topology_source_and_sink() {
 
     let (topology, _) = start_topology(config.build().unwrap(), false).await;
 
-    let event = Event::Log(LogEvent::from("this"));
+    let mut event = Event::Log(LogEvent::from("this"));
     in1.send_event(event.clone()).await.unwrap();
 
     topology.stop().await;
 
     let res = out1.flat_map(into_event_stream).collect::<Vec<_>>().await;
 
+    *event.metadata_mut().source_mut() = Some(Arc::new(OutputId::from("in1")));
```

Comment:
This pattern looks like a `fn with_source(self, source_id: Arc<OutputId>) -> Self` would be useful to reduce this to `let event = Event::Log(X).with_source(Y);`.

Reply:
Unfortunately not. The event is being used and cloned at line 145 without the source, and then the source is added at line 151 to test that the topology has added the source as well.

Reply:
But there are other places where that would be useful.